src/Metadata/Property/Factory/AnnotationPropertyNameCollectionFactory.php

```diff
@@ -63,7 +63,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
         // Properties
         foreach ($reflectionClass->getProperties() as $reflectionProperty) {
             if (null !== $this->reader->getPropertyAnnotation($reflectionProperty, ApiProperty::class)) {
-                $propertyNames[$reflectionProperty->name] = true;
+                $propertyNames[$reflectionProperty->name] = (string)$reflectionProperty->name;
```

Comment:
I'm not a fond of the centralized `array_map` solution. It looks hackish... This is not the reponsibility of `PropertyNameCollection` to fix the data provided by friend classes. Other classes should provide data respecting the contract (an array of strings).